src/blocks/remote-data-container/hooks/useRemoteData.ts

```diff
@@ -92,10 +92,21 @@ export function useRemoteData( {
 	initialPage,
 	initialPerPage,
 	initialSearchInput,
-	inputVariables = [],
 	onSuccess,
 	queryKey,
 }: UseRemoteDataInput ): UseRemoteData {
+	const blockConfig = getBlockConfig( blockName );
+
+	const query = blockConfig?.selectors?.find( selector => selector.query_key === queryKey );
+
+	if ( ! query ) {
+		throw new Error( `Selector not found for query key: ${ queryKey }` );
+	}
+
+	const inputVariables = query.inputs;
+
+	const requiredInputVariables = inputVariables.filter( input => input.required );
+
 	const [ data, setData ] = useState< RemoteData >();
```

Comment:
Nitpick: I find it helpful when hooks are kept at the top of the component. It allows you to quickly understand what is stateful.